catroid/src/test/java/org/catrobat/catroid/test/content/bricks/UpdateVariableInFormulaBrickTest.java

```diff
@@ -0,0 +1,174 @@
+		Sprite sprite = new Sprite();
+		Script script = new WhenScript();
+		formulaBrick = (FormulaBrick) scriptClass.newInstance();
+
+		userVariable.setName(VARIABLE_NAME);
+		userList.setName(VARIABLE_NAME);
+
+		project.addScene(scene);
+		scene.addSprite(sprite);
+		sprite.addScript(script);
+		script.addBrick(formulaBrick);
+
+		ProjectManager.getInstance().setCurrentProject(project);
+	}
+
+	@Parameterized.Parameter
+	public String simpleName;
+	@Parameterized.Parameter(1)
+	public Class scriptClass;
+
+	@Test
+	public void testRenamingVariable() {
+		FormulaElement element = new FormulaElement(FormulaElement.ElementType.USER_VARIABLE,
+				VARIABLE_NAME, null);
+		Formula newFormula = new Formula(element);
+		ConcurrentFormulaHashMap map = formulaBrick.getFormulaMap();
+		map.forEach((k, v) -> {
+			formulaBrick.setFormulaWithBrickField(k, newFormula);
+		});
+
+		ProjectManager.updateUserDataReferences(VARIABLE_NAME, NEW_VARIABLE_NAME, userVariable);
+
+		map.forEach((k, v) -> {
+			assertEquals(v.getTrimmedFormulaString(CatroidApplication.getAppContext()),
+					NEW_VARIABLE_USERVARIABLE_FORMAT);
+		});
+	}
+
+	@Test
+	public void testRenamingVariableNoChanges() {
+		FormulaElement element = new FormulaElement(FormulaElement.ElementType.USER_VARIABLE,
+				VARIABLE_NAME, null);
+		Formula newFormula = new Formula(element);
+		ConcurrentFormulaHashMap map = formulaBrick.getFormulaMap();
+		map.forEach((k, v) -> {
+			formulaBrick.setFormulaWithBrickField(k, newFormula);
+		});
+
+		ProjectManager.updateUserDataReferences(INVALID_NAME, NEW_VARIABLE_NAME, userVariable);
+
+		map.forEach((k, v) -> {
+			assertEquals(v.getTrimmedFormulaString(CatroidApplication.getAppContext()),
+					VARIABLE_NAME_USERVARIABLE_FORMAT);
+		});
+	}
+
+	@Test
+	public void testRenamingList() {
+		FormulaElement element = new FormulaElement(FormulaElement.ElementType.USER_LIST,
+				VARIABLE_NAME, null);
+		Formula newFormula = new Formula(element);
+		ConcurrentFormulaHashMap map = formulaBrick.getFormulaMap();
+		map.forEach((k, v) -> {
+			formulaBrick.setFormulaWithBrickField(k, newFormula);
+		});
+
+		ProjectManager.updateUserDataReferences(VARIABLE_NAME, NEW_VARIABLE_NAME, userList);
+
+		map.forEach((k, v) -> {
+			assertEquals(v.getTrimmedFormulaString(CatroidApplication.getAppContext()),
+					NEW_VARIABLE_USERLIST_FORMAT);
+		});
+	}
+
+	@Test
+	public void testRenamingListNoChanges() {
+		FormulaElement element = new FormulaElement(FormulaElement.ElementType.USER_LIST,
+				VARIABLE_NAME, null);
+		Formula newFormula = new Formula(element);
+		ConcurrentFormulaHashMap map = formulaBrick.getFormulaMap();
+		map.forEach((k, v) -> {
+			formulaBrick.setFormulaWithBrickField(k, newFormula);
+		});
+
+		ProjectManager.updateUserDataReferences(INVALID_NAME, NEW_VARIABLE_NAME, userList);
+
+		map.forEach((k, v) -> {
+			assertEquals(v.getTrimmedFormulaString(CatroidApplication.getAppContext()),
+					VARIABLE_NAME_USERLIST_FORMAT);
+		});
+	}
```

Comment:
please test that treestructure is obeyed, by using CompositeBrickTest and CompositeBrickWithSecondaryListTest as template to test all composite bricks but in this case one Formula brick, eg. Set X Brick is sufficient.